sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1331,6 +1331,16 @@ def _retry_zones(
             to_provision, 'region should have been set by the optimizer.')
         region = clouds.Region(to_provision.region)
 
+        if (not to_provision.cloud.check_quota_not_zero(
```

Comment:
Should we wrap this call to `check_quota_not_zero` in a try catch block? Something like:

```
try: 
  has_non_zero_quota = to_provision.cloud.check_quota_not_zero(....)
except:
  # In the event of unexpected quota check failures, try provisioning anyway
  has_non_zero_quota = True
```

The intent is to safeguard the provisioning pipeline from any random failures that may occur in quota checking.